operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -666,13 +666,31 @@ public static void delete(Path key) {
     private static Future<String> getPasswordAsync(SecretOperator secretOperator, String namespace, KafkaClientAuthentication auth) {
         if (auth instanceof KafkaClientAuthenticationPlain) {
             return secretOperator.getAsync(namespace, ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName())
-                    .compose(secret -> secret == null ? Future.failedFuture("Secret " + ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName() + " not found") :
-                            Future.succeededFuture(secret.getData().get(((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getPassword())));
+            .compose(secret -> {
+                if (secret == null) {
+                    return Future.failedFuture("Secret " + ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName() + " not found");
+                } else {
+                    String password = ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getPassword();
+                    if (!secret.getData().containsKey(password)) {
+                        return Future.failedFuture(String.format("Secret %s does not contain key %s", ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName(), password));
+                    }
+                    return Future.succeededFuture(secret.getData().get(password));
+                }   
+            });
         }
         if (auth instanceof KafkaClientAuthenticationScramSha512) {
             return secretOperator.getAsync(namespace, ((KafkaClientAuthenticationScramSha512) auth).getPasswordSecret().getSecretName())
-                    .compose(secret -> secret == null ? Future.failedFuture("Secret " + ((KafkaClientAuthenticationScramSha512) auth).getPasswordSecret().getSecretName() + " not found") :
-                            Future.succeededFuture(secret.getData().get(((KafkaClientAuthenticationScramSha512) auth).getPasswordSecret().getPassword())));
+            .compose(secret -> {
+                if (secret == null) {
+                    return Future.failedFuture("Secret " + ((KafkaClientAuthenticationScramSha512) auth).getPasswordSecret().getSecretName() + " not found");
+                } else {
+                    String password = ((KafkaClientAuthenticationScramSha512) auth).getPasswordSecret().getPassword();
```

Comment:
Same as above ... Maybe calling the variable `passwordKey` will make it more readable -> it will be more clear that it is just the key to it and not the password itself.

Reply:
updated